capstoneteamb/capstone360
# Late and edit

This branch will be used for changes to reviews including:
New features:
* marking reviews as late -- status: done
* allowing students to select a capstone session for reviews -- status: done
* allow students to edit reviews -- status: done

Fixes:
* Fix: checking review state before student -- status: done
* Fix: scores cannot be negative -- status: done

Clean up for review: done
Testing: tested all old and new features

Testing Guide:
In addition to general testing, there are three main features to test:

**Late Reviews Are Appropriately Marked as Such (and On Time Reviews are Not)**:
To test this feature, go through the normal processes of adding yourself for review, setting review reviewing your team, submitting your review, etc.

Check the database to see if your reviews are being marked on time or as late
Check the professor view of student reports for if late reviews are noted as such, and if on time reviews are noted as not late

A review should be late if it's submitted after the end date for the review of its type

You should check:
Midterms: on time and late
Finals: on time and late
(Note: editing reviews does not change if they are late or not)

**Capstone session selection:**
Reviews now take into account multiple capstone sessions for a student, and behave according to the selected sessions terms.

Add yourself to the number of capstone sessions you wish to test, as a professor

Then navigate to the student page

Do you see a dropdown selection menu?
Are all your capstone sessions displayed there?
---you should test one and several sessions

Does clicking 'review' take you to the correct capstone session?
Do you see the appropriate teammates?
Is the review closed (before dates), midterm (empty fields, first time), midterm (populated data, editing), final (first time, empty fields), final (populated data, editing) appropriately?
When you submit a review, are reviews submitted for the correct team and capstone session?
Is the correct student record updated for midterm/final done?'

**Editing Reviews**
To edit a review:
Go through the usual process of filling out a review
Now go back to the student dashboard
Select the same capstone session if not already selected
Click review

Are you back on the review page?
Are the data fields appropriately populated?
If you filled out a midterm review previously, but not final, and you go to final, do you see a blank review?

When you change data and submit, are the appropriate data entries changed (should only change the entries you are directly changing)?
Is the rest of the data kept the same?

Are you able to edit midterm and final reviews appropriately?

If a team member is removed, is the appropriate data still displayed for the remaining team members?

If you made a mistake, like entering the wrong points total, is data incorrectly overwritten?
